src/Validation/FiValidation.php

```diff
@@ -81,4 +82,133 @@ public static function personId($check)
 
         return $check[strlen($check) - 1] === $list[intval(substr($check, 0, 6) . substr($check, 7, 3)) % 31];
     }
+
+    /**
+     * Checks a country specific business identification number.
+     *
+     * @param string $check The value to check.
+     *
+     * @return bool Success
+     */
+    public static function businessId($check)
+    {
+        $businessId = strtoupper(trim($check));
+
+        if (strpos($businessId, '-')) {
+            if (strpos($businessId, '-') != 7) {
+                return false;
+            }
+            if (substr_count($businessId, '-') > 1) {
+                return false;
+            }
+        }
+        $businessId = str_replace('-', '', $businessId);
+
+        if (strlen($businessId) != 8) {
+            return false;
+        }
+        if (!is_numeric($businessId)) {
+            return false;
+        }
+
+        $checkResult = intval($businessId[strlen($businessId) - 1]);
+        $weights = [7, 9, 10, 5, 8, 4, 2];
+        $n = 0;
+
+        foreach (str_split(substr($businessId, 0, 7)) as $k => $y) {
+            $n += $weights[$k] * $y;
+        }
+
+        $match = $n % 11 == 0 ? 0 : 11 - $n % 11;
+        if ($match == 10) {
+            return false;
+        }
+        if ($match === $checkResult) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Checks country specific creditor reference number
+     *
+     * @param string $check The value to check.
+     *
+     * @return bool Success
+     */
+    public static function creditorReference($check)
+    {
+        if (preg_match('/^RF[0-9]{8,}$/', $check) === false) {
```

Comment:
Why does this have to be case sensitive? Wouldnt lowercase chars just be as valid from a human readable semantic point of view?

Reply:
https://en.wikipedia.org/wiki/Creditor_Reference#Structure

According to specs, creditor reference should start with "RF". If you think that "rf" should be accepted as well, I will change regex

Reply:
Well, from a human standpoint those should both be valid.
But I am fine keeping the strict one here, especially if people store it in a Postgres DB which is also cs by default.

Reply:
@dereuromark Sorry, is there something I still need to change in order to get it ready for merge?